x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/JobUpdate.java

```diff
@@ -517,6 +541,7 @@ boolean isNoop(Job job) {
                 && (dailyModelSnapshotRetentionAfterDays == null
                         || Objects.equals(dailyModelSnapshotRetentionAfterDays, job.getDailyModelSnapshotRetentionAfterDays()))
                 && (resultsRetentionDays == null || Objects.equals(resultsRetentionDays, job.getResultsRetentionDays()))
+                && (annotationsRetentionDays == null || Objects.equals(annotationsRetentionDays, job.getAnnotationsRetentionDays()))
```

Comment:
I guess it can be set to a very high value to effectively unset it.

The `isNoop(job)` function is only used in tests and should probably be moved to `JobUpdateTests`. Not necessary for this PR though.

Reply:
> But this prevents annotationsRetentionDays from ever being unset once it is set.

So maybe we should consider keeping the wire type signed and use -1 to encode "unset"?

> The isNoop(job) function is only used in tests and should probably be moved to JobUpdateTests. Not necessary for this PR though.

Ack